pkg/image/validation/registry_test.go

```diff
@@ -0,0 +1,238 @@
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"The 'name' field is required for each entry in 'images'.",
+			},
+		},
+		`duplicate name`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				ContainerImages: []image.ContainerImage{
+					{
+						Name: "foo",
+					},
+					{
+						Name: "bar",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "baz",
+					},
+					{
+						Name: "bar",
+					},
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"Duplicate image name 'foo' found in the 'images' section.",
+				"Duplicate image name 'bar' found in the 'images' section.",
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ear := test.Registry
+			failures := validateContainerImages(&ear)
+			assert.Len(t, failures, len(test.ExpectedFailedMessages))
+
+			var foundMessages []string
+			for _, foundValidation := range failures {
+				foundMessages = append(foundMessages, foundValidation.userMessage)
+				assert.Equal(t, registryComponent, foundValidation.component)
+			}
+
+			for _, expectedMessage := range test.ExpectedFailedMessages {
+				assert.Contains(t, foundMessages, expectedMessage)
+			}
+		})
+	}
+}
+
+func TestValidateHelmCharts(t *testing.T) {
+	tests := map[string]struct {
+		Registry               image.EmbeddedArtifactRegistry
+		ExpectedFailedMessages []string
+	}{
+		`no helm charts`: {
+			Registry: image.EmbeddedArtifactRegistry{},
+		},
+		`valid charts`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				HelmCharts: []image.HelmChart{
+					{
+						Name:    "foo",
+						RepoURL: "http://valid.com", // shows http:// is allowed
+						Version: "1.0",
+					},
+					{
+						Name:    "bar",
+						RepoURL: "https://valid.com", // shows https:// is allowed
+						Version: "2.0",
+					},
+				},
+			},
+		},
+		`missing fields`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				HelmCharts: []image.HelmChart{
+					{},
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"The 'name' field is required for each entry in 'charts'.",
+				"The 'repoURL' field is required for each entry in 'charts'.",
+				"The 'version' field is required for each entry in 'charts'.",
+			},
+		},
+		`duplicate chart`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				HelmCharts: []image.HelmChart{
+					{
+						Name:    "foo",
+						RepoURL: "http://foo.com",
+						Version: "1.0",
+					},
+					{
+						Name:    "foo",
+						RepoURL: "https://bar.com",
+						Version: "2.0",
+					},
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"Duplicate chart name 'foo' found in the 'charts' section.",
+			},
+		},
+		`invalid repo`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				HelmCharts: []image.HelmChart{
+					{
+						Name:    "foo",
+						RepoURL: "example.com",
+						Version: "1.0",
+					},
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"The 'repoURL' field must begin with either 'http://' or 'https://'.",
+			},
+		},
+	}
```

Comment:
Same nit suggestion regarding individual failure cases (which is probably not necessary to be honest). Though I will say that I really like the way you concatenate the error messages, very cool.